src/StdAssertions.sol

```diff
@@ -322,4 +322,55 @@ abstract contract StdAssertions is DSTest {
             assertApproxEqRelDecimal(a, b, maxPercentDelta, decimals);
         }
     }
+
+    function assertEqCall(address target, bytes memory callDataA, bytes memory callDataB) internal virtual {
+        assertEqCall(target, callDataA, target, callDataB, true);
+    }
+
+    function assertEqCall(address targetA, bytes memory callDataA, address targetB, bytes memory callDataB)
+        internal
+        virtual
+    {
+        assertEqCall(targetA, callDataA, targetB, callDataB, true);
+    }
+
+    function assertEqCall(address target, bytes memory callDataA, bytes memory callDataB, bool strictRevertData)
+        internal
+        virtual
+    {
+        assertEqCall(target, callDataA, target, callDataB, strictRevertData);
+    }
+
+    function assertEqCall(
+        address targetA,
+        bytes memory callDataA,
+        address targetB,
+        bytes memory callDataB,
+        bool strictRevertData
+    ) internal virtual {
+        (bool successA, bytes memory returnDataA) = address(targetA).call(callDataA);
+        (bool successB, bytes memory returnDataB) = address(targetB).call(callDataB);
+
+        if (successA && successB) {
+            assertEq(returnDataA, returnDataB, "Call return data does not match");
+        }
+
+        if (!successA && !successB && strictRevertData) {
+            assertEq(returnDataA, returnDataB, "Call revert data does not match");
+        }
+
+        if (!successA && successB) {
+            emit log("Error: Call reverted unexpectedly");
```

Comment:
Maybe we change this and the branch below to
```suggestion
            emit log("Error: Calls were not equal");
```

The current message implies that A was intended to succeed because B did, but I don't think that's true (i.e. maybe they both should have reverted).

Reply:
I had first followed the convention assertEq(testValue, referenceValue), but then saw that forge-std changed this to Left/Right.